hexrd/transforms/new_capi/xf_new_capi.py

```diff
@@ -28,142 +29,364 @@
+    """
+
+    Takes triplets of angles in the beam frame (2*theta, eta, omega)
+    to components of unit diffraction vectors in the LAB frame.  If the omega
+    values are not trivial (i.e. angs[:, 2] = 0.), then the components
+    are in the SAMPLE frame.  If the crystal rmat is specified and
+    is not the identity, then the components are in the CRYSTAL frame.
+
+
+    Parameters
+    ----------
+    angs : ndarray
+        The euler angles of diffraction. The last dimension must be 3.  In
+        (2*theta, eta, omega) format.
+    beam_vec : ndarray, optional
+        Unit vector pointing towards the X-ray source in the lab frame.
+        Defaults to [0,0,-1]
+    eta_vec : ndarray, optional
+        Vector defining eta=0 in the lab frame.  Defaults to [1,0,0]
+    chi : float, optional
+        The inclination angle of the sample frame about the lab frame X.
+    rmat_c : ndarray, optional
+        The change of basis matrix from the reciprocal frame to the crystal
+        frame.  Defaults to the identity.
+
+    Returns
+    -------
+    ndarray
+        (3,n) array of unit diffraction vectors, frame depends on the input
+        parameters
+    """
     # TODO: Improve capi to avoid multiplications when rmat_c is None
     beam_vec = beam_vec if beam_vec is not None else cnst.beam_vec
     eta_vec = eta_vec if eta_vec is not None else cnst.eta_vec
 
-    angs = np.ascontiguousarray( np.atleast_2d(angs) )
-    beam_vec = np.ascontiguousarray( beam_vec.flatten() )
-    eta_vec = np.ascontiguousarray( eta_vec.flatten() )
-    rmat_c = np.ascontiguousarray(rmat_c) if rmat_c is not None else cnst.identity_3x3
+    angs = np.ascontiguousarray(np.atleast_2d(angs))
+    beam_vec = np.ascontiguousarray(beam_vec.flatten())
+    eta_vec = np.ascontiguousarray(eta_vec.flatten())
+    rmat_c = (
+        np.ascontiguousarray(rmat_c)
+        if rmat_c is not None
+        else cnst.identity_3x3
+    )
     chi = 0.0 if chi is None else float(chi)
 
-    return _impl.anglesToDVec(angs,
-                                         beam_vec, eta_vec,
-                                         chi, rmat_c)
+    return _impl.anglesToDVec(angs, beam_vec, eta_vec, chi, rmat_c)
+
 
 def makeGVector(hkl, bMat):
+    """
+    Take a crystal relative b matrix onto a list of hkls to output unit
+    reciprocal latice vectors (a.k.a. lattice plane normals)
+
+
+    Parameters
+    ----------
+    hkl : ndarray
+        (3,n) ndarray of n hstacked reciprocal lattice vector component
+        triplets
+    bMat : ndarray
+        (3,3) ndarray of the change of basis matrix from the reciprocal
+        lattice to the crystal reference frame
+
+    Returns
+    -------
+    ndarray
+        (3,n) ndarray of n unit reciprocal lattice vectors (a.k.a. lattice
+        plane normals)
+
+    """
     assert hkl.shape[0] == 3, 'hkl input must be (3, n)'
     return unitVector(np.dot(bMat, hkl))
```

Comment:
Undefined referenec here. Should be unit_vector, but I didn't change it because unit_vector works as the transpose of unitVector (so you either change unit_vector or do unitVector(np.dot(bMat, hkl).T).T). Also it's not part of the API